src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -459,6 +463,12 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                         sendEmptyMessageDelayed(FLUSH_QUEUE, mTrackEngageRetryAfter);
                         mFailedRetries++;
                         logAboutMessageToMixpanel("Retrying this batch of events in " + mTrackEngageRetryAfter + " ms");
+                        break;
+                    }
+
+                    eventsData = dbAdapter.generateDataString(table);
+                    if (eventsData != null) {
+                        queueCount = Integer.valueOf(eventsData[2]);
```

Comment:
(we use magic index numbers all over the place in this file eg. https://github.com/mixpanel/mixpanel-android/pull/336/files#diff-7c00d8762ab328cad50eba1fecb26bc5R402    not very readable)